From bbd4090207f21615b96ab563921c6e306df9b870 Mon Sep 17 00:00:00 2001
From: Tom Zanussi <tom.zanussi@linux.intel.com>
Date: Thu, 26 May 2022 12:36:59 -0700
Subject: [PATCH 1387/1418] crypto: iax: Break out of loop after list del in
 del_iax_wq()

Once the element has been deleted, there's no reason to continue
iterating the list, just break out.

Signed-off-by: Tom Zanussi <tom.zanussi@linux.intel.com>
---
 drivers/crypto/iax/iax_crypto_main.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/iax/iax_crypto_main.c b/drivers/crypto/iax/iax_crypto_main.c
index 02e2e88adb3a..0143bbf4e026 100644
--- a/drivers/crypto/iax/iax_crypto_main.c
+++ b/drivers/crypto/iax/iax_crypto_main.c
@@ -413,10 +413,9 @@ static void del_iax_wq(struct iax_device *iax_device, struct idxd_wq *wq)
 
 			pr_debug("%s: removed wq %p from iax_device %p, n_wq %d, nr_iax %d\n", __func__, wq, iax_device, iax_device->n_wq, nr_iax);
 
-			if (iax_device->n_wq == 0) {
+			if (iax_device->n_wq == 0)
 				del_iax_device(iax_device);
-				break;
-			}
+			break;
 		}
 	}
 }
-- 
2.31.1

